LEGO1/omni/src/stream/mxdsbuffer.cpp

```diff
@@ -254,26 +258,18 @@ MxResult MxDSBuffer::ParseChunk(
 		MxU32 length = p_header->GetLength() + MxDSChunk::GetHeaderSize() + 8;
 		MxDSBuffer* buffer = new MxDSBuffer();
 
-		if (buffer && buffer->AllocateBuffer(length, e_allocate) == SUCCESS &&
-			buffer->CalcBytesRemaining((MxU8*) p_data) == SUCCESS) {
-			*p_streamingAction = new MxDSStreamingAction((MxDSStreamingAction&) *p_action);
-
-			if (*p_streamingAction) {
-				MxU16* flags = MxStreamChunk::IntoFlags(buffer->GetBuffer());
-				*flags = p_header->GetChunkFlags() & ~DS_CHUNK_SPLIT;
-
-				delete p_header;
-				(*p_streamingAction)->SetUnknowna0(buffer);
-				goto done;
-			}
-		}
-
-		if (buffer) {
+		if (!buffer || buffer->AllocateBuffer(length, e_allocate) || buffer->CalcBytesRemaining((MxU8*) p_data) ||
```

Comment:
I think we should test for != SUCCESS on both function calls here so it's clearer